plugins/brain/__init__.py

```diff
@@ -1784,6 +1784,417 @@ def _get_target_view(ctx, target):
+        response = find_exact_duplicates(ctx)
+
+        if not ctx.delegated:
+            ctx.trigger("reload_dataset")
+
+        return response
+
+    def resolve_output(self, ctx):
+        outputs = types.Object()
+        outputs.str(
+            "num_total_dups",
+            label="Number of samples with exact duplicates",
+        )
+        outputs.str(
+            "num_unique_dups",
+            label="Number of exact duplicate groupings",
+        )
+        view = types.View(label="Exact duplicate results")
+        return types.Property(outputs, view=view)
+
+
+def find_exact_duplicates_inputs(ctx, inputs):
+    get_target_view(ctx, inputs)
+
+
+def find_exact_duplicates(ctx):
+    target = ctx.params.get("target", None)
+
+    dataset = ctx.dataset
+    target_view = _get_target_view(ctx, target)
+
+    duplicates_dict = fob.compute_exact_duplicates(target_view)
+
+    rep_ids = []
+    dup_ids = []
+    for rep_id, neighbor_ids in duplicates_dict.items():
+        rep_ids.append(rep_id)
+        dup_ids.append(rep_id)
+        dup_ids.extend(neighbor_ids)
+
+    dups_view = dataset.select(dup_ids, ordered=True)
+    reps_view = dataset.select(rep_ids)
+
+    dataset.save_view("exact duplicates", dups_view, overwrite=True)
+    dataset.save_view(
+        "representatives of exact duplicates", reps_view, overwrite=True
+    )
+
+    return {
+        "num_total_dups": len(dup_ids),
+        "num_unique_dups": len(rep_ids),
+    }
+
+
+class DeduplicateExactDuplicates(foo.Operator):
+    @property
+    def config(self):
+        return foo.OperatorConfig(
+            name="deduplicate_exact_duplicates",
+            label="Deduplicate exact duplicates",
+            description=(
+                "Delete all but one copy from each group of exact duplicates"
+            ),
+            light_icon="/assets/icon-light.svg",
+            dark_icon="/assets/icon-dark.svg",
+            allow_delegated_execution=True,
+            allow_immediate_execution=True,
+            default_choice_to_delegated=True,
+            dynamic=True,
+        )
+
+    def resolve_input(self, ctx):
+        inputs = types.Object()
+
+        dataset = ctx.dataset
+        if dataset.has_saved_view("exact duplicates"):
+            dups_view = dataset.load_saved_view("exact duplicates")
+            reps_view = dataset.load_saved_view(
+                "representatives of exact duplicates"
+            )
+
+            warning = types.Warning(
+                label=(
+                    f"Your last exact duplicates scan detected {len(dups_view)} "
+                    f"exact duplicates across {len(reps_view)} groupings. "
+                    "Executing this operator will delete all but one sample "
+                    "from each group."
+                )
+            )
+            inputs.view("warning", warning)
+        else:
+            deduplicate_exact_duplicates_inputs(ctx, inputs)
+
+            warning = types.Warning(
```

Comment:
In the case where there is no existing exact duplicates scan, I added a warning explaining what triggering this method will do:

<img width="1157" height="210" alt="Screenshot 2025-07-24 at 2 15 48 AM" src="https://github.com/user-attachments/assets/5e4a0168-a383-4d27-8a15-ef19d4b15cb8" />